CCompiler/Token.cs

```diff
@@ -5,17 +5,26 @@ public class Token
     public readonly TokenType type;
     public readonly string text;
 
+    /// <summary>
+    /// origin index in file , do not have this in Equals
+    /// </summary>
```

Comment:
Should we use XML comments? Whats the benefit?

Reply:
looks good in visual studio intellisense